engine.py

```diff
@@ -912,3 +912,58 @@ def __setup_favorite_dirs(self):
                                 type=(nuke.IMAGE | nuke.SCRIPT | nuke.GEO),
                                 icon=icon_path,
                                 tooltip=path)
+
+    #####################################################################################
+    # Script and session related methods
+
+    def get_session_path(self):
+        """
+        Returns the path to the current file if it resides on
+        disk. If unsaved, it returns None.
```

Comment:
I think the comment is potentially misleading - it sounds to me like we are saying 'if there are unsaved changed, it returns None'. maybe just to be clear, it could say something like 'if the session has never been saved and isn't associated with a file on disk yet, `None` is returned.